### PR TITLE
Updated content security policy to allow script and style from google analytics

### PR DESCRIPTION
**Pull request by RajeshPaul38**
Signed-off-by: Rajesh Paul <rajesh.paul@progress.com>

### Description

[Please describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by RajeshPaul38** [comment]
Buildkite: https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/423

**Comment by RajeshPaul38** [comment]
ad hoc build has passed

File: src/supermarket/app/views/application/_analytics.html.erb [hunk 1]
```diff
@@ -8,15 +8,13 @@
...
-     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-     })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){window.dataLayer.push(arguments);}
+    gtag('js', new Date());
 
-    ga('create', '<%= ENV["GOOGLE_ANALYTICS_ID"] %>', 'auto');
-    ga('require', 'linkid', 'linkid.js');
-    ga('require', 'displayfeatures');
-    ga('send', 'pageview');
```
Comment (RajeshPaul38) [inline src/supermarket/app/views/application/_analytics.html.erb:20]: This snippet doesn't seem to be working. Got the reference to the new script from this link: https://developers.google.com/analytics/devguides/collection/gtagjs

**Comment by msys-sgarg** [comment]
@RajeshPaul38  Can we have screenshots with the PR to make sure that google analytics are working fine with this change ?

**Comment by RajeshPaul38** [comment]
> @RajeshPaul38 Can we have screenshots with the PR to make sure that google analytics are working fine with this change ?

Sure, let me attach

**Comment by RajeshPaul38** [comment]
Here is how the data is getting reflected in google analytics. @msys-sgarg 
Screenshot 2022-03-25 at 1 45 35 PM

**Review by antima-gupta — COMMENTED** [review]
Looks good to me

**Review by dheerajd-msys — APPROVED** [review]

**Comment by msys-sgarg** [comment]
@RajeshPaul38 I hope we took a look into the script injection possibilities with this change ?

**Comment by RajeshPaul38** [comment]
@msys-sgarg thanks for pointing out the possible chances of XSS attack. I've tried to mitigate it as much as possible by disabling inline scripts and only keeping the inline css enabled. I've used a CSP mechanism called nonce which ensures only the inline scripts that is signed with a valid nonce, will be allowed by the CSP. This nonce is a hash that gets generated uniquely for every request. Hence the chances of XSS attack is zero as no one else will know what will be the nonce hash for a certain request. Hope this is a satisfactory resolution for the issues pointed out during the demo. If yes pls approve.

**Comment by RajeshPaul38** [comment]
Build: https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/438

**Comment by RajeshPaul38** [comment]
Checked my changes in an EC2 instance. See below screenshot:

Screenshot 2022-03-30 at 1 26 20 AM

**Comment by RajeshPaul38** [comment]
Can we get this merged with priority @saghoshprogress as this is a customer bug?

**Comment by RajeshPaul38** [comment]
Ready for merge @saghoshprogress

**Comment by RoyShravani** [comment]
@RajeshPaul38  Deploy preview check failed. please take a look!

**Comment by RajeshPaul38** [comment]
I've tried debugging the issue with netlify build. But seems like an issue in the build process, not related to this PR. Pls approve.

**Pull request merged**